halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/SpringProfileFactory.java

```diff
@@ -41,6 +43,22 @@ protected void setProfile(
 
     profile.appendContents(
         yamlToString(deploymentConfiguration.getName(), profile, spectatorConfig));
+
+    // Front50 contains persistent storage configuration under the spinnaker top level key, so we
+    // add extensibility configuration for Front50 in Front50ProfileFactory
+    if (this.getClass() != Front50ProfileFactory.class) {
```

Comment:
Instead of introspecting to figure out what class this is, shouldn't this use an overriden method instead?  In general it's a bad pattern for a parent class to switch on what the implementing class is.

Reply:
ah, so you're saying to remove this check and update Front50ProfileFactory to override setProfile? That makes sense to me and would be cleaner 👍

Reply:
Yes, exactly, or even something `boolean includeExtensibilityConfig()` that defaults to true in the base class but is overriden to false in `Front50` would also work, and this block could check that flag.

Reply:
I like it!